website/client/js/controllers/filtersCtrl.js

```diff
@@ -17,6 +17,7 @@ habitrpg.controller("FiltersCtrl", ['$scope', '$rootScope', 'User', 'Shared',
             User.updateTag({params:{id:tag.id}, body:{name:tag.name}});
         })
         $scope._editing = false;
+        User.sync();
```

Comment:
In fact, doing it this way will cause a race condition to occur. If the user sync completes before the updateTag operation completes, it'll remove the update from the ui until _another_ sync.